### PR TITLE
Add support for expressions in the line directive

### PR DESCRIPTION
**Pull request by JCapucho**
Based on the `IfParser` adds support for parsing the line and source file id as expressions

Also has a very small commit to fix clippy lints in the latest stable has those were making the pipelines have to much noise to be meaningful

### CODE REVIEW

File: src/pp_tests.rs [hunk 1]
```diff
@@ -986,17 +983,16 @@ fn parse_line() {
...
-    // );
-
     // test that the integer must fit in a u32
     check_preprocessing_error("#line 4294967296u", PreprocessorError::LineOverflow);
     // test that the integer must fit in a u32
     check_preprocessed_result("#line 4294967295u", "");
+    check_preprocessing_error("#line -1", PreprocessorError::LineOverflow);
+
+    // Test some expression
+    check_preprocessed_result("#line 20 << 2 + 1", "");
```
Comment (Kangz) [inline src/pp_tests.rs]: Maybe we could add a __LINE__ on a new line and check that it produces the correct number?

**Review by Kangz — APPROVED** [review]
LGTM!

**Comment by JCapucho** [comment]
It should be ready to be merged now

**Review by Kangz — APPROVED** [review]
LGTM

**Pull request merged**